app/models/address.rb

```diff
@@ -135,8 +135,23 @@ def to_a
   end
 
   def fetch_county_fips_code
-    us_county = BenefitMarkets::Locations::CountyFips.where(state_postal_code: state, county_name: county).first
-    us_county ? us_county.county_fips_code : nil
+    state_abbrev = state.blank? ? '' : state.upcase
+    county_name = county.blank? ? '' : county.titlecase
+    county_fip = ::BenefitMarkets::Locations::CountyFips.where(state_postal_code: state_abbrev, county_name: county_name).first
+
+    if county_fip.present?
+      county_fip.county_fips_code
+    else
+      county_fip_by_zip_state.present? ? county_fip_by_zip_state.county_fips_code : ""
+    end
+  end
+
+  def county_fip_by_zip_state
+    state_abbrev = state.blank? ? '' : state.upcase
+    zip_code = zip.blank? ? "" : zip.scan(/\d+/).join
+    counties = ::BenefitMarkets::Locations::CountyZip.where(:zip => zip_code, :state => state_abbrev)
+    county_name = counties.count == 1 ? counties.first.county_name.titlecase : ""
+    ::BenefitMarkets::Locations::CountyFips.where(state_postal_code: state_abbrev, county_name: county_name).first
```

Comment:
If we have more than one county-zip pair returned won't the this search fail anyway?  Why even run the search in such a case?

Reply:
when we imported data for ME, in some instances we are missing county data for address, so we imported address without the county name. And we have enrollments associated with these accounts.

So my thought here is, for a state and zip code combination if we find only one county, we can find the fips code using this county and state and include it in enrollment EDI so that enrollment won't error for the fips code missing error at the carrier end.